### PR TITLE
Fixes #16694 - Add endpoint for Arf report downloads

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -55,7 +55,11 @@
             get 'content'
           end
         end
-        resources :arf_reports, :only => [:index, :show, :destroy]
+        resources :arf_reports, :only => [:index, :show, :destroy] do
+          member do
+            get 'download'
```
Comment (ares) [inline config/routes.rb]: I suggest we disable this cop

File: config/routes.rb [hunk 2]
```diff
@@ -55,7 +55,11 @@
             get 'content'
           end
         end
-        resources :arf_reports, :only => [:index, :show, :destroy]
+        resources :arf_reports, :only => [:index, :show, :destroy] do
```
Comment (ares) [inline config/routes.rb]: I suggest we disable this cop, both syntax are fine

File: app/controllers/api/v2/compliance/arf_reports_controller.rb [hunk 3]
```diff
@@ -62,12 +62,31 @@ def create
           render :json => { :result => :OK, :id => arf_report.id.to_s }
         end
 
+        api :GET, "/compliance/arf_reports/:id/download/", N_("Download bzipped ARF report")
```
Comment (ares) [inline app/controllers/api/v2/compliance/arf_reports_controller.rb]: I suggest we disable this cop

File: config/routes.rb (hunk 2 above)
Comment (xprazak2) [inline config/routes.rb]: I believe we have it [disabled](https://github.com/theforeman/foreman_openscap/blob/master/.rubocop.yml#L15-L17), is it possible hound is ignoring it?

File: config/routes.rb (hunk 1 above)
Comment (xprazak2) [inline config/routes.rb]: Fine with me.

**Comment by xprazak2** [comment]
I rebased and disabled the StringLiterals rule, hound seems to be satisfied.

**Comment by ares** [comment]
One more thing is missing, you need to list the action for at least one permission, probably here - https://github.com/theforeman/foreman_openscap/blob/master/lib/foreman_openscap/engine.rb#L53

**Comment by xprazak2** [comment]
I always seem to forget permissions...

**Comment by shlomizadok** [comment]
We are missing test coverage for this. Will you please add?

**Comment by xprazak2** [comment]
I rebased and added a test case.

**Comment by shlomizadok** [comment]
Merging. Thanks a lot, @xprazak2 !

**Pull request merged**